src/XMLSecurityDSig.php

```diff
@@ -772,6 +772,15 @@ public function locateKey($node=null)
     }
 
     /**
+     * Returns:
+     *  1 on succesful signature verification,
+     *  0 when signature verification failed,
+     *  -1 if an error occurred during processing.
+     *
+     * NOTE: be very careful when checking the return value, because in PHP,
+     * -1 will be cast to True when in boolean context. So always check the
+     * return value in a strictly typed way, e.g. "$obj->verify(...) === 1".
+     *
```

Comment:
Not always true as the underlying verifySignature method can return bool when its an hmac sha1 signature.